extract_indices.ipynb

```diff
@@ -0,0 +1,366 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "b7ee45fd-5069-4cfc-9e3a-a7cdb79753b3",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Kaktovik done\n",
+      "Stevens Village done\n",
+      "Igiugik Village done\n",
+      "Levelock done\n",
+      "Eyak done\n",
+      "Ketchikan done\n",
+      "Aleutians done\n"
+     ]
+    }
+   ],
+   "source": [
+    "with pd.ExcelWriter(idx_decade_summary_fp, engine=\"openpyxl\") as decade_writer:\n",
+    "    dfs = []\n",
+    "    for location in locations:\n",
+    "        lat, lon = locations[location]\n",
+    "\n",
+    "        df_rows = []\n",
+    "\n",
+    "        for decade in decades:\n",
+    "            start_year, end_year = decade.split(\"-\")\n",
+    "            year_sl = slice(int(start_year), int(end_year))\n",
+    "            for index in index_list:\n",
+    "                if decade in [\"1980-1989\", \"1990-1999\"]:\n",
+    "                    # this will be the historical scenario\n",
+    "                    scenario = \"hist\"\n",
+    "                    for model in ds.model.values:\n",
+    "                        da = subset_data(ds, index, model, scenario, year_sl, lat, lon)\n",
+    "                        df_rows.append(summarize_to_row(da, model, scenario, decade, index))\n",
+    "\n",
+    "                elif decade == \"2000-2009\":\n",
+    "                    # mixed decade, do both and concatenate data arrays\n",
+    "                    hist_sl = slice(2000, 2005)\n",
+    "                    future_sl = slice(2006, 2009)\n",
+    "                    for model in ds.model.values:\n",
+    "                        for scenario in scenarios:\n",
+    "                            hist_da = subset_data(ds, index, model, \"hist\", year_sl, lat, lon)\n",
+    "                            future_da = subset_data(ds, index, model, scenario, year_sl, lat, lon)\n",
+    "                            da = xr.concat([hist_da, future_da], dim=\"year\")\n",
+    "                            df_rows.append(summarize_to_row(da, model, scenario, decade, index))\n",
+    "\n",
+    "                else:\n",
+    "                    # future scenarios\n",
+    "                    for model in ds.model.values:\n",
+    "                        for scenario in scenarios:\n",
+    "                            da = subset_data(ds, index, model, scenario, year_sl, lat, lon)\n",
+    "                            df_rows.append(summarize_to_row(da, model, scenario, decade, index))\n",
+    "\n",
+    "        # create dataframe write dataframe to a sheet in the excel file   \n",
+    "        df = pd.DataFrame(df_rows)\n",
```

Comment:
The same rounding issue appears here, too. Might need to round the DataFrame after conversion:

```
pd.DataFrame(df_rows).round(1)
```

Reply:
Great catch and thanks this seems to work! Fixed, and fixed comments/typos as suggested in all the other comments as well - going to resolve those and merge.